src/modules/dashboard/components/ManageDomainsDialog/ManageDomainsDialog.tsx

```diff
@@ -0,0 +1,69 @@
+import React from 'react';
+import { defineMessages } from 'react-intl';
+
+import { DialogProps } from '~core/Dialog';
+import IndexModal from '~core/IndexModal';
+
+import { WizardDialogType } from '~utils/hooks';
+
+const MSG = defineMessages({
+  dialogHeader: {
+    id: 'dashboard.ManageDomainsDialog.dialogHeader',
+    defaultMessage: 'Manage domains',
+  },
+  createNewDomainTitle: {
+    id: 'dashboard.ManageDomainsDialog.createNewDomainTitle',
+    defaultMessage: 'Create new domain',
+  },
+  createNewDomainDescription: {
+    id: 'dashboard.ManageDomainsDialog.createNewDomainDescription',
+    defaultMessage:
+      'Teams, departments, circles: domains let you froup types of activity.',
+  },
+  editDomainTitle: {
+    id: 'dashboard.ManageDomainsDialog.editDomainTitle',
+    defaultMessage: 'Edit domain',
+  },
+  editDomainDescription: {
+    id: 'dashboard.ManageDomainsDialog.editDomainDescription',
+    defaultMessage: `Need to repurpose domain? Here's the place to do it.`,
+  },
+});
+
+interface CustomWizardDialogueProps {
+  nextStep: string;
+  prevStep: string;
+}
+
+type Props = DialogProps & WizardDialogType<object> & CustomWizardDialogueProps;
+
+const displayName = 'dashboard.ManageDomainsDialog';
+
+const ManageDomainsDialog = ({ cancel, close, callStep, prevStep }: Props) => {
```

Comment:
This could have also set up the permissions check for the domains, but I guess it's also ok to leave it like this.

On the other hand, you'll have to add a permissions check for domains in your current #2339 so might as well add it here as well.

Up to you.